src/math/fft/fft.c

```diff
@@ -5,14 +5,21 @@
 // Author: Amery Song <chao.song@intel.com>
 //	   Keyon Jie <yang.jie@linux.intel.com>
 
-#include <sof/audio/coefficients/fft/twiddle_32.h>
 #include <sof/audio/buffer.h>
 #include <sof/audio/format.h>
 #include <sof/common.h>
 #include <sof/lib/alloc.h>
 #include <sof/math/fft.h>
 
-struct fft_plan *fft_plan_new(struct icomplex32 *inb, struct icomplex32 *outb, uint32_t size)
+#if CONFIG_MATH_16BIT_FFT
```

Comment:
I think I need to split fft.c into separate 16 bit and 32 bit modules and use cmakelist.txt to select based on kconfig the sources. This allows cmocka test to build e.g. 32 bit library and test it even if platform configuration has disabled it. I ran into problem with auditory library. It depends on FFT and it won't build if CONFIG_MATH_xxBIT_FFT are not defined. When there are no #if CONFIGs in module they can be always built to unit test.

Reply:
Splitting is good, as it allows Kconfig to select for size vs quality.